lib/cn/cn_metrics.h

```diff
@@ -148,7 +148,9 @@ cn_ns_samp(const struct cn_node_stats *ns)
     u64 alen = cn_ns_alen(ns);
     u64 clen = cn_ns_clen(ns);
 
-    return clen ? 100 * alen / clen : 0;
+    assert(clen);
+
+    return clen ? 100 * alen / clen : 100;
```

Comment:
Did you find a case where clen == 0 ?  I'm guessing tombs don't figure into the clen calculation?

Reply:
ns->ns_hclen was never really being set, so it was always zero. And in kvsets with only ptombs, ns_kclen and ns_vclen are zero. Which makes clen = ns_hclen + ns_kclen + ns_vclen = 0.